cg/meta/demultiplex/validation.py

```diff
@@ -57,14 +60,11 @@ def is_flow_cell_ready_for_delivery(flow_cell_directory: Path) -> bool:
     return Path(flow_cell_directory, DemultiplexingDirsAndFiles.DELIVERY).exists()
 
 
-def validate_sample_sheet_exists(flow_cell_run_directory: Path) -> None:
-    sample_sheet_path: Path = Path(
-        flow_cell_run_directory, DemultiplexingDirsAndFiles.SAMPLE_SHEET_FILE_NAME
-    )
+def validate_sample_sheet_exists(flow_cell: FlowCellDirectoryData) -> None:
+    sample_sheet_path: Path = flow_cell.get_sample_sheet_path_hk()
     if not sample_sheet_path.exists():
-        raise FlowCellError(
-            f"Sample sheet {sample_sheet_path} does not exist in flow cell run directory."
-        )
+        raise FlowCellError(f"Sample sheet {sample_sheet_path} does not exist in housekeeper.")
+    LOG.debug(f"Found sample sheet {sample_sheet_path} in housekeeper.")
```

Comment:
I think just `if not sample_sheet_path:` should be enough. This risks running into `None type object has not attribute exists()`.

Reply:
I agree that we should have the null check but I do not think this is sufficient - would this approach not miss if the file on Hasta was missing although the db entry is filled in? 

I would prefer `if (not sample_sheet_path or not sample_sheet_path.exists())` and use the lazy loading of "or" in Python.

Reply:
fixed to check for both.